### PR TITLE
Orchestrate shardnet restaking

### PR DESCRIPTION
**Pull request by nikurt**
Uses `tools/restaked` to ensure a validator node re-stakes their tokens after being kicked.
Has an option to also create the initial validator accounts.
Assumes `shardnet-bootX` are the boot nodes.

### CODE REVIEW

File: pytest/tests/shardnet/restake.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+import mocknet
+from configured_logger import logger
+
+
+def restart_restaked_script(restaked_url, delay_sec):
+    s = '''
+        killall restaked
+        wget '{restaked_url}' -O /home/ubuntu/restaked
```
Comment (mina86) [inline pytest/tests/shardnet/restake.py:16]: No need for quotes, shlex.quote will take care of it.  In fact, with quotes this is invalid.

File: pytest/tests/shardnet/restake.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import pathlib
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+import mocknet
+from configured_logger import logger
+
+
+def restart_restaked_script(restaked_url, delay_sec):
+    s = '''
+        killall restaked
```
Comment (mina86) [inline pytest/tests/shardnet/restake.py:15]: `killall restaked; sleep 1; killall -9 restaked; rm -f /home/ubuntu/restaked`? -9 to make sure it’s killed, `rm` to make sure we don’t get ETXTBSY if killall takes its sweet time before wget opens the file.

File: scripts/check_pytests.py [hunk 3]
```diff
@@ -23,9 +23,10 @@
 # is covered here as well since those tests are not run on NayDuck any more.
 HELPER_SCRIPTS = [
     'delete_remote_nodes.py',
+    'loadtest/*',
     'mocknet/*',
+    'shardnet/*',
     'stress/hundred_nodes/*',
-    'loadtest/*',
```
Comment (mina86) [inline scripts/check_pytests.py:28]: I’m beginning to think we need to move all of those into a different directory. pytest/tools perhaps?

File: pytest/tests/shardnet/scripts/create_account.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+account_id=$(cat /home/ubuntu/.near/shardnet/validator_key.json  | grep account_id | awk -F'"' '{print $4}')
```
Comment (mina86) [inline pytest/tests/shardnet/scripts/create_account.sh:2]: Useless use of cat. `grep account_id /home/ubuntu/.near/shardnet/validator_key.json`. Same on line 5.

**Review by mina86 — APPROVED** [review]

File: scripts/check_pytests.py (hunk 3 above)
Comment (mm-near) [reply scripts/check_pytests.py:28]: +1 for 'python tools' directory somewhere

File: pytest/tests/shardnet/restake.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import argparse
```
Comment (mm-near) [inline pytest/tests/shardnet/restake.py:1]: please add README.md file to this directory (with command line examples)

File: pytest/tests/shardnet/restake.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import shlex
+import random
+import sys
+from rc import pmap
+import pathlib
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
```
Comment (mm-near) [inline pytest/tests/shardnet/restake.py:8]: not related to this PR -- @mina86 - is there a better solution for this? could we set some 'env' instead?

File: pytest/tests/shardnet/restake.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        wget '{restaked_url}' -O /home/ubuntu/restaked
+        nohup ./restaked --home /home/ubuntu/.near/shardnet/ --rpc-url 127.0.0.1:3030 --wait-period {delay_sec} --stake-amount {stake_amount} 1>>/home/ubuntu/restaked.out 2>>/home/ubuntu/restaked.err </dev/null &
+    '''.format(restaked_url=shlex.quote(restaked_url),
+               stake_amount=shlex.quote(str(random.randint(10**3, 10**5))),
+               delay_sec=shlex.quote(str(delay_sec)))
+    logger.info(f'Starting restaked: {s}')
+    return s
+
+
+def create_account(node, near_pk, near_sk):
```
Comment (mm-near) [inline pytest/tests/shardnet/restake.py:25]: I'm surprised that this executes the code on the remote machine.. please add comments (either here or on the top of file)

File: pytest/tests/shardnet/scripts/create_account.sh (hunk 4 above)
Comment (mm-near) [inline pytest/tests/shardnet/scripts/create_account.sh:2]: what account are you creating?

File: pytest/tests/shardnet/scripts/create_account.sh [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (mm-near) [inline pytest/tests/shardnet/scripts/create_account.sh:1]: ok - now I see, 

so this tool is used to create the validator account (for the keys that are already present on the machine) - and you have to pass the private&public key for 'near' user as arguments.

optional: I'd probably use node.machine.upload to upload the near.json to the machines instead.

**Review by mm-near — APPROVED** [review]

**Pull request merged**